tests/large/test_swift.py

```diff
@@ -0,0 +1,78 @@
+
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014 Canonical
+#
+# Authors:
+#  Didier Roche
+#  Tin Tvrtković
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; version 3.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+"""Tests for the Swift category"""
+import subprocess
+import os
+import tempfile
+from tests.large import LargeFrameworkTests
+from tests.tools import UMAKE, spawn_process
+
+
+class SwiftTests(LargeFrameworkTests):
+    """The default Swift compiler."""
+
+    TIMEOUT_INSTALL_PROGRESS = 300
+
+    EXAMPLE_PROJECT = """print("Hello, world!")"""
+
+    def setUp(self):
+        super().setUp()
+        self.installed_path = os.path.join(self.install_base_path, "swift", "swift-lang")
+        self.framework_name_for_profile = "Swift Lang"
+
+    @property
+    def exec_path(self):
+        return os.path.join(self.installed_path, "usr", "bin", "swift")
+
+    def test_default_swift_install(self):
+        """Install Swift from scratch test case"""
+        if not self.in_container:
+            self.example_prog_dir = tempfile.mkdtemp()
+            self.additional_dirs.append(self.example_prog_dir)
+            example_file = os.path.join(self.example_prog_dir, "Package.swift")
+            open(example_file, "w").write("")
+            os.mkdir(os.path.join(self.example_prog_dir, "Sources"))
+            example_file = os.path.join(self.example_prog_dir, "Sources", "main.swift")
+            open(example_file, "w").write(self.EXAMPLE_PROJECT)
+            compile_command = ["bash", "-l", "-c", "swift build"]
+        else:  # our mock expects getting that path
```

Comment:
You don't provide a path here contrary to the go one. ;)
I would say "expects getting that command parameter"